control-panel

ebraheemabbas opened the pull request:
add control-panel page with two btn
add admin level to the control-panel pages
closes #46

ebraheemabbas committed:
control-panel
add control-panel page with two btn
add admin level to the control-panel pages
closes #46

okarem commented:
i'm having issues running it, it says data is not defined, in header component, do i need to create my own jwt object?

jamiecoe commented:
@okarem you need to have a fake jwt cookie in your browser. You can create one by going to Application tab (in dev tools), and manually adding one like this:

<img width="601" alt="Screen Shot 2019-12-23 at 12 36 31" src="https://user-images.githubusercontent.com/18463718/71353142-e0749600-2580-11ea-9767-17db73f98ca7.png">

Just put some random strings/numbers for `jwt` Value - it doesn't actually matter.

jamiecoe commented:
@ebraheemabbas some styling issues here - the word 'Storage' should be inline with 'Print New' and 'Control Panel'. Also, Control Panel has it's own icon (see here #22).

<img width="362" alt="Screen Shot 2019-12-23 at 12 35 14" src="https://user-images.githubusercontent.com/18463718/71353221-11ed6180-2581-11ea-8bb7-4e12014454b8.png">

okarem committed:
fixed it so it wont break the whole app
the work in this branch is not complete, please add a dummy Admin user token to the fake server, otherwise the whole work will come to a halt after merging

jamiecoe commented on src/components/Sensors-action-page/Sensors-actions.jsx:
```diff
@@ -33,6 +32,16 @@ export default function SensorsActionPage() {
+                history.push("/control-panel");
+              }}
+            >
+              <img src={StorageIcon} alt="Control panel" />
```
Use the Control Panel icon here, not the Storage icon (see #22)

jamiecoe commented on src/components/Sensors-action-page/Sensors-actions.jsx:
```diff
@@ -33,6 +32,16 @@ export default function SensorsActionPage() {
               <span>Storage</span>
             </button>
           </div>
+          <div className="button-container">
```
We need some logic here so that ONLY admin users will see the Control Panel button on the home page. 

There's no point displaying the Control Panel button to normal users - if they try to click it, they will be immediately kicked out.

jamiecoe commented on src/components/general/ProtectedRoute/ProtectedRoute.jsx:
```diff
@@ -3,14 +3,17 @@ import { Route, useHistory } from "react-router-dom";
-     return;
+  if (
+    data.isAuthenticated === false ||
+    (adminLevel !== data.admin && adminLevel === true)
```
We could refactor this line to be clearer:
```
    (adminLevel !== data.admin && adminLevel === true)
```

You care about whether `adminLevel` is `true` AND `data.admin` is `false`, so why not have:
```
(adminLevel === true && data.admin === false)
```

jamiecoe commented on src/components/general/header.jsx:
```diff
@@ -14,7 +14,7 @@ export default function Header() {
         <div className="nav-items">
           <Clock className="Clock" ticking={true} format={"L HH:mm"} />
-          <div className="userName">{data.isAuthenticated === false ? null :data.employee}</div>
+          <div className="userName">{(data&&data.isAuthenticated) ? data.employee:null}</div>
```
```suggestion
          <div className="userName">{(data && data.isAuthenticated) ? data.employee : null}</div>
```

ebraheemabbas committed:
resolve some change req

jamiecoe commented:
@ebraheemabbas 

Having the employee name in the header is causing it to be wider than the iPad screen:

<img width="490" alt="Screen Shot 2019-12-23 at 14 51 05" src="https://user-images.githubusercontent.com/18463718/71359103-e627a700-2593-11ea-8d0d-fb4d3d4e7b17.png">

Can you fix it so the header fits into the screen width?

ebraheemabbas committed:
give the navbar 100% width
